src/plonk_util.rs

```diff
@@ -312,10 +312,76 @@ pub fn halo_g<F: Field>(x: F, us: &[F]) -> F {
     product
 }
 
+/// Computes the commitments (without randomness) `Comm(L_i)` for `i=0..num`, where `L_i` is the `i-th`
+/// Lagrange basis polynomial on the subgroup of order `order` of `C`'s scalar field,
+/// i.e., `L_i(g^j) = delta_ij`, with `g` a `order`-th primitive root of unity.
+pub fn precompute_lagrange_commitments<C: Curve>(
+    order_log: usize,
+    num: usize,
+    msm_precomputation: &MsmPrecomputation<C>,
+) -> Vec<ProjectivePoint<C>> {
```

Comment:
There's probably a better way to compute these. Here I just use the naive way of computing the Lagrange basis polynomial coefficients one by one.
Also this could be data that we can store in a file and use it as a trustless (since anyone can recompute it) SRS.